Psychtoolbox/PsychDemos/DrawFormattedText2Demo.m

```diff
@@ -0,0 +1,138 @@
+% DrawFormattedText2Demo
+%
+% Draws formatted text, shows how to align and center text vertically
+% and/or horizontally, how line-breaks are introduced, how to transform
+% text.
+%
+% Press any key to cycle through different demo displays.
+%
+% see also: PsychDemos, Screen DrawText?, DrawSomeTextDemo, DrawFormattedText2Demo
+
+% 09/15/16    dcn    Wrote it.
+
+try
+    AssertOpenGL;   % check PTB is functioning
+    
+    % Choosing the display with the highest display number is
+    % a best guess about where you want the stimulus displayed.
+    screens=Screen('Screens');
+    screenNumber=max(screens);
+
+    % Open window with default settings:
+    PsychImaging('PrepareConfiguration');
+    % make screen partially transparent on OSX and windows vista or
+    % higher, so we can debug.
+%    PsychDebugWindowConfiguration;
+    [w,wrect]=PsychImaging('OpenWindow', screenNumber, 128);
+    
+    % This preference setting selects the high quality text renderer 
+    % each operating system: It is not really needed, as the high quality
+    % renderer is the default on all operating systems, so this is more of
+    % a "better safe than sorry" setting. DrawFormattedText2 _only_ works
+    % with this renderer. So it fails to load (see the warnings on the
+    % command line, the output of this demo will look horrible)
+    Screen('Preference', 'TextRenderer', 1);
+    Screen('BlendFunction', w, GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
```

Comment:
If you don't use that Screen('Preference', 'TextAlphaBlending', 1); setting, Screen DrawText will automatically select GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA alpha-blending, temporarily overriding your 'Blendfunction', as that setting is the one needed in all the "normal" 99.99% of all cases of text drawing for proper text anti-aliasing (which just uses alpha on the contours of glyphs to properly fade them out into the background). If you set that flag, it will use the regular 'Blendfunction', hopefully with the results you wanted by carefully designing your drawing code. So at default settings that 'Blendfunction' neither helps or hurts, it is just redundant.